test/performance/README.md

```diff
@@ -0,0 +1,49 @@
+# Performance tests
+
+This directory contains performance tests for `Knative Serving`:
+
+- Tests are using [kperf](https://github.com/knative-sandbox/kperf); a testing tool designed to run load tests in knative
+
+## Prerequisites
+
+- If running the tests using `go test` then you need to install kperf since the tests assumes its already installed.
+- Running `performance-tests.sh` installs kperf in the temporary folder to enable the ci to run the tests.
+- to install kperf:
+  - run `git clone https://github.com/knative-sandbox/kperf.git`
+  - run `make build`
+
+## Flags
+
+- These flags are used by the performance tests framework used here:
+  - `service-prefix`: The prefix for the services generated by kperf.
+  - `services-count` The count of Services generated by kperf.
+  - `kperf-output`: kperf output file(s) location.
+  - `service-average`: The average time tests should expect for the kperf service creation.
+  - `service-timeout`: The time the tests should wait for the kperf Service creation timeout.
+
+- Flags required for accessing GCS bucket to either upload results, or compare with previously upload results:
```

Comment:
Looks like this pulls in quite a lot of dependencies.. wdyt about removing the upload stuff for now and focusing the initial PR on just running a single, simple kperf test? That should make this one (and the follow ons) easier to review, and we can think a bit more about exactly how upload and comparison should work in the follow-on.

Reply:
my concern was that without comparison to a previous or nightly run, the results would not be meaningful

Reply:
I don't know if I follow why we are even using go test here in that case? If the only goal is to generate and compare output - and I am indeed now noticing the tests don't really have any assertions as opposed to just error cases.. - what's the value in using go test at all? To ask in another way: why don't we just directly run the kperf binary (in the prow job) and output/upload the results?

Reply:
FWIW I think a perfectly good MVP that would deliver a ton of value would be "run the perf tests as an optional suite in prow that can be requested with a comment (like the optional Istio mesh tests are), output as text in the prow job". That would let a contributor making a change quickly run a reasonably representative performance test on a roughly standard environment before and after a change (with a bit of manual work to pull out the data from the prow log, but that's fine to begin with) to establish if it was an improvement/regression. Uploading and comparison over time would be awesome after that, but aren't something I personally see as required to get us started.

Reply:
sorry I'm not following, are you saying the comparison in not needed at this point and we should only have a bash script running from prow to just run kperf, there are a few things kperf doesn't do like setup the cluster and create the namespaces and wait for the services to scale down, so I thought it could be better to do that in go code in the form of a test instead of bash. The optional prow job is already there now but its runs these tests.

Reply:
I'm saying a couple things:

1) For an MVP I think just a single run without comparison is fine (actually super valuable), and I'd really appreciate a smaller PR to review 😅 
2) If we're not actually planning to make any real assertions, go test seems maybe a bit odd/overkill to use just in order to set up a cluster etc

And I guess a bonus thing:

3) Have we considered moving the stuff happening in the "test" here (creating the ksvc, for example) in to kperf instead? I'm not sure where the boundary between kperf and serving should be, but at first pass I'm not sure I follow why kperf doesn't have a mode where it sets up the service. That would seem to make kperf more useful, and would avoid us having lots of kperf-specific code (as opposed to just a prow action to run kperf and output/upload results) in serving. Are there downsides to that I'm not seeing?

Reply:
1- if you think a single run generating a file in prow without any comparison is enough, that fine, I didn't think it would be enough to show
2- I'm hoping to add assertions, but wasn't sure what numbers to consider as a cutoff for a test failure. 
3- the creation of the services is in kperf, but the setup of the cluster and namespaces and waiting for the service to scale down isn't, we can add it there, don't know where to draw the line between what's in kperf and what's here, but if we decide serving will not do anything except run the kperf binary, then we can move everything in kperf, I need to take another look to see if there is any limitations there.

Reply:
1- yeah, I think it'd be more than enough - it'd be great
1- I suspect this is a bit similar to Bench* tests: you theoretically _could_ try to make them fail in a yes/no way, but it's pretty useful just to be able to compare two runs qualitatively by looking at the benchmark output before and after (and over time). To put it another way I think perf tests are more about helping guide/validate improvements and judge them, and less a red/green yes/no regression protection sort of thing
1- That's interesting, I think we don't have any namespace creation in the regular e2e test go code today (you have to run `ko apply -f test/config` to set up the namespace, or the e2e tests [run ytt]( https://github.com/knative/serving/blob/main/test/e2e-common.sh#L252)). In terms of waiting for a service to exit my instinct is if kperf creates the service, it should be up to kperf to clean it up.